components/messenger.rst

```diff
@@ -86,6 +86,65 @@ that will do the required processing for your message::
        }
     }
 
+Envelope
+--------
+
+The notion of an envelope is a concept that helps add context around the
+messages. An envelope is a message and a set of data. From a user's perspective, this
+allows you to set some configuration around the message. For example, to set the serialization
+groups used when the message goes through the transport layer, wrap your message
+in an ``Envelope`` and add some ``SerializerConfiguration``::
+
+    use Symfony\Component\Messenger\Envelope;
+    use Symfony\Component\Messenger\Transport\Serialization\SerializerConfiguration;
+
+    $bus->dispatch(
+        (new Envelope($message))->with(new SerializerConfiguration([
+            'groups' => ['my_serialization_groups'],
+        ]))
+    );
+
+At the moment, the Symfony Messenger has the following built-in envelopes:
+
+1. :class:`Symfony\\Component\\Messenger\\Transport\\Serialization\\SerializerConfiguration`,
+   to configure the serialization groups used by the transport.
+2. :class:`Symfony\\Component\\Messenger\\Middleware\\Configuration\\ValidationConfiguration`,
+   to configure the validation groups used when the validation middleware is enabled.
+3. :class:`Symfony\\Component\\Messenger\\Asynchronous\\Transport\\ReceivedMessage`,
+   an internal item that marks the message as received from a transport.
+
+Instead of dealing directly with the messages in the middleware you can receive the
+envelope by implementing the :class:`Symfony\\Component\\Messenger\\EnvelopeAwareInterface`
+marker, like this::
+
+    use Symfony\Component\Messenger\Asynchronous\Transport\ReceivedMessage;
+    use Symfony\Component\Messenger\Middleware\MiddlewareInterface;
+    use Symfony\Component\Messenger\EnvelopeAwareInterface;
+
+    class MyOwnMiddleware implements MiddlewareInterface, EnvelopeAwareInterface
+    {
+        public function handle($message, callable $next)
```

Comment:
Actually, maybe we can rename `$message` to `$envelope` in this example? PHP allows us to do so (i.e. variable names not to match with the name in the interface). It might be clearer.

Reply:
Yes. Please rename this.